app.rb

```diff
@@ -1,8 +1,78 @@
-print "Which calculator would you like to use? Type B for basic or A for advanced"
-which_calc = gets.chomp
-if which_calc == "B"
-  puts "What operation would you like to perform? Choose (a)dd, (s)ubstract, (m)ultiply or (d)ivide."
+begin # do while loop
+  print "Which calculator would you like to use? Type B for basic, A for advanced, or BMI to calculate your BMI. Type exit to exit the app."
+  which_calc = gets.chomp
+  if which_calc == "B"
```

Comment:
These options are a bit precise, I would allow for upper or lowercase. I would also avoid long inputs like imperial because it increases the chance the user will input something incorrect.